raiden/transfer/node.py

```diff
@@ -691,16 +714,10 @@ def handle_processed(
                         message.balance_proof.transferred_amount,
                         message.recipient,
                     ))
-                remove.append(pos)
 
-        for removepos in reversed(remove):
-            queue.pop(removepos)
-
-        if not queue:
-            queueids_to_remove.append(queueid)
-
-    for queueid in queueids_to_remove:
-        del chain_state.queueids_to_queues[queueid]
+    # Clean up message queue
+    for queueid in list(chain_state.queueids_to_queues.keys()):
```

Comment:
Please, don't clean all queues! `handle_processed` should clear only the specific queue `QueueIdentifier(recipient, channel_identifier)`), and `handle_delivered` should clear only the global queue QueueIdentifier(recipient, CHANNEL_IDENTIFIER_GLOBAL_QUEUE)`. As it's done now, you'll clear ordered queue upon `Delivered` which will lead to errors if the client is offline or doesn't handle correctly the message on first receiving it (retry logic)

Reply:
Oh, nvm, `Processed` has precedence over `Delivered`, and on `Delivered` you only clear global messages.

Reply:
This does no clear all queues, please refer to the `inplace_delete_message_queue` source code. @rakanalh perhaps we can improve the name of the function?

Reply:
I saw it, the problem was not in the `inplace_delete_message_queue` logic, but in the for that runs it upon `Processed`, but as a `Processed` message **was** Delivered, it's no problem to clear it here, and I didn't see it before, that's why I already approved it.